PyInstaller/building/utils.py

```diff
@@ -603,30 +602,19 @@ def get_code_object(modname, filename, optimize):
     return code_object
 
 
-def strip_paths_in_code(co, new_filename=None):
-    # Paths to remove from filenames embedded in code objects
-    replace_paths = sys.path + CONF['pathex']
-    # Make sure paths end with os.sep and the longest paths are first
-    replace_paths = sorted((os.path.join(f, '') for f in replace_paths), key=len, reverse=True)
-
-    if new_filename is None:
-        original_filename = os.path.normpath(co.co_filename)
-        for f in replace_paths:
-            if original_filename.startswith(f):
-                new_filename = original_filename[len(f):]
-                break
-
-        else:
-            return co
-
-    code_func = type(co)
+def replace_filename_in_code_object(code_object, filename):
+    """
+    Recursively replace the `co_filename` in the given code object and code objects stored in its `co_consts` entries.
+    Primarily used to anonymize collected code objects, i.e., by removing the build environment's paths from them.
+    """
+    code_type = type(code_object)  # built-in <class 'code'>
```

Comment:
There's `types.CodeType` if that's any easier (personally I'm not fussed).

Reply:
Yeah, I find `types.CodeType` a bit nicer, even though it's the really the same thing under-the-hood.